[
   {
      "body" : "Your Issue title does not reflect your problem. `ERROR: AcceptToMemoryPool` happens during normal operations (although I think we should change it from ERROR to INFO).\r\n\r\n`2016-06-08 13:57:59 Corruption: block checksum mismatch` is more important.\r\n\r\nYour node has shutdown because of a corrupted block file. Either you have force terminated your node in the wrong moment or your Disk/Storage-Device is malfunctioning.",
      "created_at" : "2016-06-09T06:44:18Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8174#issuecomment-224813106",
      "id" : 224813106,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8174",
      "updated_at" : "2016-06-09T06:44:18Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/224813106",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/178464?v=3",
         "events_url" : "https://api.github.com/users/jonasschnelli/events{/privacy}",
         "followers_url" : "https://api.github.com/users/jonasschnelli/followers",
         "following_url" : "https://api.github.com/users/jonasschnelli/following{/other_user}",
         "gists_url" : "https://api.github.com/users/jonasschnelli/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/jonasschnelli",
         "id" : 178464,
         "login" : "jonasschnelli",
         "organizations_url" : "https://api.github.com/users/jonasschnelli/orgs",
         "received_events_url" : "https://api.github.com/users/jonasschnelli/received_events",
         "repos_url" : "https://api.github.com/users/jonasschnelli/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/jonasschnelli/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/jonasschnelli/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/jonasschnelli"
      }
   },
   {
      "body" : "> Your Issue title does not reflect your problem. ERROR: AcceptToMemoryPool happens during normal operations (although I think we should change it from ERROR to INFO).\r\n\r\nThis is solved in master.\r\nSee also #7592",
      "created_at" : "2016-06-09T07:54:53Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8174#issuecomment-224824838",
      "id" : 224824838,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8174",
      "updated_at" : "2016-06-09T07:58:07Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/224824838",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/126646?v=3",
         "events_url" : "https://api.github.com/users/laanwj/events{/privacy}",
         "followers_url" : "https://api.github.com/users/laanwj/followers",
         "following_url" : "https://api.github.com/users/laanwj/following{/other_user}",
         "gists_url" : "https://api.github.com/users/laanwj/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/laanwj",
         "id" : 126646,
         "login" : "laanwj",
         "organizations_url" : "https://api.github.com/users/laanwj/orgs",
         "received_events_url" : "https://api.github.com/users/laanwj/received_events",
         "repos_url" : "https://api.github.com/users/laanwj/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/laanwj/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/laanwj/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/laanwj"
      }
   },
   {
      "body" : "I'm having this issue on FreeBSD with ZFS storage (so no on-disk data corruption is possible). I've already performed reindexing, but after several days of working in background, bitcoind crashed again.\r\n\r\nBitcoin Core Daemon version v0.12.1.0-g9779e1e (built from ports).\r\n\r\n```\r\n2016-07-12 06:33:54 UpdateTip: new best=000000000000000004d423b7951f1d2db204ed70eadf0a17125f9d315f734372  height=420315  log2_work=84.972631  tx=141779952  date=2016-07-11 18:39:09 progress=0.999735  cache=57.6MiB(36867tx)\r\n2016-07-12 06:33:54 UpdateTip: 5 of last 100 blocks have unexpected version\r\n2016-07-12 06:33:58 UpdateTip: new best=0000000000000000026d11266fe61e9cc64b458c84152673d50d1c8e0d0abb88  height=420316  log2_work=84.972666  tx=141781031  date=2016-07-11 18:44:01 progress=0.999737  cache=59.5MiB(39496tx)\r\n2016-07-12 06:33:58 UpdateTip: 5 of last 100 blocks have unexpected version\r\n2016-07-12 06:33:59 Corruption: block checksum mismatch\r\n2016-07-12 06:33:59 *** System error while flushing: Database corrupted\r\n2016-07-12 06:33:59 Error: Error: A fatal internal error occurred, see debug.log for details\r\n2016-07-12 06:33:59 ERROR: ProcessNewBlock: ActivateBestChain failed\r\n2016-07-12 06:33:59 tor: Thread interrupt\r\n2016-07-12 06:33:59 scheduler thread interrupt\r\n2016-07-12 06:33:59 torcontrol thread exit\r\n2016-07-12 06:33:59 addcon thread interrupt\r\n2016-07-12 06:33:59 msghand thread interrupt\r\n2016-07-12 06:33:59 net thread interrupt\r\n2016-07-12 06:34:03 opencon thread interrupt\r\n2016-07-12 06:34:03 Shutdown: In progress...\r\n2016-07-12 06:34:03 StopNode()\r\n2016-07-12 06:34:04 Corruption: block checksum mismatch\r\n2016-07-12 06:34:04 *** System error while flushing: Database corrupted\r\n2016-07-12 06:34:04 Error: Error: A fatal internal error occurred, see debug.log for details\r\n2016-07-12 06:34:04 Shutdown: done\r\n```",
      "created_at" : "2016-07-12T08:35:58Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8174#issuecomment-231974195",
      "id" : 231974195,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8174",
      "updated_at" : "2016-07-12T08:35:58Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/231974195",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/946343?v=3",
         "events_url" : "https://api.github.com/users/vadipp/events{/privacy}",
         "followers_url" : "https://api.github.com/users/vadipp/followers",
         "following_url" : "https://api.github.com/users/vadipp/following{/other_user}",
         "gists_url" : "https://api.github.com/users/vadipp/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/vadipp",
         "id" : 946343,
         "login" : "vadipp",
         "organizations_url" : "https://api.github.com/users/vadipp/orgs",
         "received_events_url" : "https://api.github.com/users/vadipp/received_events",
         "repos_url" : "https://api.github.com/users/vadipp/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/vadipp/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/vadipp/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/vadipp"
      }
   },
   {
      "body" : "I'm getting various types of database corruption errors (mostly just like this one) when trying to synchronize bitcoin-qt after quite a long time (~one year) without running it. One of the failures even broke my wallet.dat (no idea how that could happen) to the point where I had to restore it from backups.\r\n\r\nThe machine runs Btrfs, so silent data corruption is highly unlikely. (Btrfs is a checksummed filesystem, just like ZFS.) The machine doesn't have any software issues other than the bitcoin problem and passes days of memtesting without failure.\r\n\r\nBTW, `--reindex` doesn't solve the problem; it simply fails again sooner or later in the reindexing process. The only thing that does help is `bitcoin-qt -reindex -rpcthreads=1`. This leads me to believe that there's an ugly race condition lurking somewhere in bitcoin and/or bitcoin-qt.\r\n\r\nHowever, I don't know why the tweak above helped and whether it was just a coincidence, because IIUC, the `-rpcthreads` option would only set the number of threads serving requests from the frontend(s) and should not have any direct impact on reindexing. And indeed, the number of threads that run during the reindexing proces is still the same. Yet it just doesn't fail with `-rpcthreads=1`.\r\n\r\nMy system: ArchLinux, kernel 4.6.4 (vanilla kernel with a homebrew configuration, not the distro kernel), bitcoin-qt 0.12.1-1, gcc-multilib 6.1.1-3.",
      "created_at" : "2016-07-19T22:01:27Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8174#issuecomment-233780118",
      "id" : 233780118,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8174",
      "updated_at" : "2016-07-19T22:01:27Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/233780118",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/947737?v=3",
         "events_url" : "https://api.github.com/users/andrejpodzimek/events{/privacy}",
         "followers_url" : "https://api.github.com/users/andrejpodzimek/followers",
         "following_url" : "https://api.github.com/users/andrejpodzimek/following{/other_user}",
         "gists_url" : "https://api.github.com/users/andrejpodzimek/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/andrejpodzimek",
         "id" : 947737,
         "login" : "andrejpodzimek",
         "organizations_url" : "https://api.github.com/users/andrejpodzimek/orgs",
         "received_events_url" : "https://api.github.com/users/andrejpodzimek/received_events",
         "repos_url" : "https://api.github.com/users/andrejpodzimek/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/andrejpodzimek/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/andrejpodzimek/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/andrejpodzimek"
      }
   }
]
